Extend stack_overflow_protection for lcd_move_e and _lcd_level_bed

Blue-Marlin opened the pull request:
Extend stack_overflow_protection for lcd_move_e() and _lcd_level_bed() with
a refined method of #3050.
Simple, small, local, elegant.

Fix for one of AnatoliyKube's problems in MarlinFirmware#3054 (comment) (if  i solved his language riddle correct)

thinkyhead commented on Marlin/ultralcd.cpp:
```diff
@@ -862,7 +862,8 @@ static void lcd_move_e(
+    if (movesplanned() <= 3)
+      line_to_current(E_AXIS);
     lcdDrawUpdate = 1;
   }
```
As this is, it will drop moves, and those moves will stay dropped when the LCD times out and returns to the info screen. However, it turns out there is an easy solution here to avoid dropping any moves:

``` cpp
if (encoderPosition != 0 && movesplanned() <= 3) {
  current_position[E_AXIS] += float((int)encoderPosition) * move_menu_scale;
  line_to_current(E_AXIS);
  encoderPosition = 0;
  lcdDrawUpdate = 1;
}
```

This holds onto the encoder position and doesn't reset it, so the next time this is invoked, the move will still be pending. With this change you can throw away the `line_to_current` call with `LCD_CLICKED` below.

thinkyhead commented on Marlin/ultralcd.cpp:
```diff
@@ -2249,13 +2253,15 @@ char* ftostr52(const float& x) {
-      line_to_current(Z_AXIS);
+      if (movesplanned() <= 3)
+        line_to_current(Z_AXIS);
       lcdDrawUpdate = 2;
```
As above, so below…
```cpp
if (max_software_endstops) NOMORE(current_position[Z_AXIS], Z_MAX_POS);
if (movesplanned() <= 3) {
  line_to_current(Z_AXIS);
  encoderPosition = 0;
  lcdDrawUpdate = 2;
}

thinkyhead commented on Marlin/ultralcd.cpp:
```diff
@@ -2249,13 +2253,15 @@ char* ftostr52(const float& x) {
+      if (movesplanned() <= 3)
+        line_to_current(Z_AXIS);
       lcdDrawUpdate = 2;
     }
```
Same idea here…

``` cpp
if (encoderPosition != 0 && movesplanned() <= 3) {
  refresh_cmd_timeout();
  current_position[Z_AXIS] += float((int)encoderPosition) * MBL_Z_STEP;
  if (min_software_endstops) NOLESS(current_position[Z_AXIS], Z_MIN_POS);
  if (max_software_endstops) NOMORE(current_position[Z_AXIS], Z_MAX_POS);
  line_to_current(Z_AXIS);
  encoderPosition = 0;
  lcdDrawUpdate = 2;
}
```

thinkyhead commented:
The same technique I suggest can also be applied to the other move routines. You can amend this PR and add those also.

Blue-Marlin committed:
Extend stack_overflow_protection for lcd_move_e and _lcd_level_bed

Extend stack_overflow_protection for lcd_move_e() and _lcd_level_bed() with
a refined method of 3050.
